pokemon/damage.js

```diff
@@ -29,82 +29,120 @@ function damage(pPokemon1,pPokemon2){
 }
 
 
+
+function getEffectivenessPoints(level){
+  var point;
+   function isHighEffectiveness () {
+     return point = 2;
+   }
+   function isNeutralEffectiveness () {
+     return point = 1;
+   }
+   function isLittleEffectiveness () {
+     return point = 0.5;
+   }
+
+   var points = {
+     'High' : isHighEffectiveness,
+     'Neutral' : isNeutralEffectiveness,
+     'Little' : isLittleEffectiveness,
+   };
+
+   return points[level]();
+}
+
+
+
 function effective(pTypePokemon1,pTypePokemon2){
-  switch(pTypePokemon1){
-    case 'fire':
+  var pokemonEffectiveness
+
+  var pokemonType = {
+    'fire' : function () {
       if(pTypePokemon2 == 'grass'){
```

Comment:
your update is good, but I think you can do it better
here you have 5 "if" statements per pokemonType, it means that potentially you have 5 execution times per request, can we apply the same Object literals solution for these 5 ifs?
something like:
`pokemonType[pTypePokemon1][pTypePokemon2]`
or
`pokemonType[pTypePokemon + pTypePokemon2]`